dev/tasks/matlab/github.yml

```diff
@@ -0,0 +1,171 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+{% import 'macros.jinja' as macros with context %}
+
+{{ macros.github_header() }}
+
+jobs:
+
+  ubuntu:
+    name: AMD64 Ubuntu 20.04 MATLAB
+    runs-on: ubuntu-latest
+    steps:
+      {{ macros.github_checkout_arrow()|indent }}
+      - name: Install ninja-build
+        run: sudo apt-get install ninja-build
+      - name: Install MATLAB
+        uses: matlab-actions/setup-matlab@v1
+        with:
+          release: R2023a
+      - name: Install ccache
+        run: sudo apt-get install ccache
+      - name: Setup ccache
+        shell: bash
+        run: arrow/ci/scripts/ccache_setup.sh
+      - name: ccache info
+        id: ccache-info
+        shell: bash
+        run: echo "cache-dir=$(ccache --get-config cache_dir)" >> $GITHUB_OUTPUT
+      - name: Cache ccache
+        uses: actions/cache@v3
+        with:
+          path: ${{ '{{' }} steps.ccache-info.outputs.cache-dir {{ '}}' }}
+          key: matlab-ccache-ubuntu-${{ '{{' }} hashFiles('arrow/cpp/**', 'arrow/matlab/**') {{ '}}' }}
+          restore-keys: matlab-ccache-ubuntu-
```

Comment:
Due to the way actions/cache's security model is setup it doesn't work to cache the ccache in the same way we do in apche/arrow. That's why we use sccache there. We have a jinja macro that sets the right envvars etc. and a script to install it. 
You can checkout the java jars job on how to set it up on all three OSes `dev/tasks/java-jars/github.yml`

Reply:
Thanks for explaining this! We were puzzled as to why the `ccache` step didn't seem to be working as expected, so that clarifies things!